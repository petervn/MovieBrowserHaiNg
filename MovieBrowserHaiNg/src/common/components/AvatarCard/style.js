import {StyleSheet} from 'react-native';
const style = StyleSheet.create({
  container: {
    margin: 5,
  },
  avatar: {
    width: 80,
    height: 80,
    borderRadius: 40,
  },
});

export default style;
