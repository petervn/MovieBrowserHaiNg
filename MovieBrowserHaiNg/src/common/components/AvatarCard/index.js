import React from 'react';
import {Text, Image, TouchableOpacity} from 'react-native';
import {getImage} from '../../utils';
import style from './style';

export default AvartarCard = ({data, onPress}) => {
  const {profile_path, name = '--'} = data;
  return (
    <TouchableOpacity
      style={style.container}
      onPress={() => {
        onPress(data);
      }}>
      <Image
        source={getImage(profile_path)}
        style={style.avatar}
        sizeMode="contain"
      />
      <Text style={{textAlign: 'center'}} numberOfLines={1}>
        {name.split(' ')[0]}
      </Text>
    </TouchableOpacity>
  );
};
