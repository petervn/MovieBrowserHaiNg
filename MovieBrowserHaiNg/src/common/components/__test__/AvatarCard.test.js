import React from 'react';
import {Text} from 'react-native';
import {shallow, mount} from 'enzyme';
import AvatarCard from '../AvatarCard';

describe('Avatar Card', () => {
  describe('Rendering', () => {
    const data = {
      profile_path: 'imgage.com/i1.png',
      name: 'Hai Ng',
    };

    it('should match to snapshot', () => {
      const component = shallow(<AvatarCard data={data} onPress={() => {}} />);
      expect(component).toMatchSnapshot();
    });

    it('should render first name', () => {
      const component = mount(<AvatarCard data={data} onPress={() => {}} />);
      const textValue = component.find(Text).text();
      expect(textValue).toEqual('Hai');
    });
  });
});
