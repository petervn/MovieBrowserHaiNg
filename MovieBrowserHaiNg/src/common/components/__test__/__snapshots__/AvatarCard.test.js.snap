// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Card Rendering should match to snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "margin": 5,
    }
  }
>
  <Image
    sizeMode="contain"
    source={
      Object {
        "uri": "https://image.tmdb.org/t/p/w500imgage.com/i1.png",
      }
    }
    style={
      Object {
        "borderRadius": 40,
        "height": 80,
        "width": 80,
      }
    }
  />
  <Text
    numberOfLines={1}
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Hai
  </Text>
</TouchableOpacity>
`;
